src/main/bootstrap/mod.rs

```diff
@@ -430,6 +440,58 @@ mod tests {
         }
     }
 
+    mod update_peers {
+        use super::*;
+
+        #[test]
+        fn it_filters_out_duplicates() {
+            let mut peers: Vec<PeerInfo> = (0..10)
+                .map(|i| peer_info_with_rand_key(ipv4_addr(1, 2, 3, 4, i)))
+                .collect();
+
+            let (tx, rx) = mpsc::channel();
+
+            let duplicates: Vec<PeerInfo> = peers.iter().take(5).map(|peer| peer.clone()).collect();
+            tx.send(duplicates).unwrap();
+
+            update_peers(&mut peers, rx);
+
+            let peers_len = peers.len();
+            let mut peers_set = HashSet::new();
+
+            assert!(peers.into_iter().all(|peer| peers_set.insert(peer)));
+            assert_eq!(peers_len, peers_set.len());
+        }
+
+        #[test]
+        fn it_preserves_order_if_shuffled_again() {
```

Comment:
When I read 
```
update_peers
    it_preserves_order_if_shuffled_again
```
It sounds like `update_peers()` has something to do with the peer shuffling... actually it does not.
I feel like the test would be a lot simpler, if we didn't include all that shuffle code. `update_peers()` simply appends peers from `rx: Receiver` to the given `Vec` and does not change the order of already inserted peers. Maybe that's what you could test? :)

Reply:
Sure, I will update it. At first I thought it would be important to see that even after appending to the list, if we shuffle it again we would have the wanted order. But since `peers` still is a `Vec<PeerInfo>` you are right, there is no need to test for that here, it is up to `shuffled_bootstrap_peers`.